File: archive/o/objective-c/fizz-buzz.m
#import <Foundation/Foundation.h>

int main(int argc, const char* argv[]) {
    @autoreleasepool {
        NSMutableString* output = [NSMutableString stringWithCapacity:8];

        for (NSInteger i = 1; i <= 100; i++) {
            [output setString:@""];

            if (i % 3 == 0) [output appendString:@"Fizz"];
            if (i % 5 == 0) [output appendString:@"Buzz"];

            if (output.length > 0) {
                printf("%s\n", output.UTF8String);
            } else {
                printf("%ld\n", (long)i);
            }
        }
    }
    return 0;
}

File: archive/l/lua/fizz-buzz.lua
for i = 1, 100 do
    output = ""

    if i % 3 == 0 then
        output = output .. "Fizz"
    end

    if i % 5 == 0 then
        output = output .. "Buzz"
    end

    if output == "" then
        output = i
    end

    print(output)
end
